### ISSUE TITLE
Can't unmask form fields

### ISSUE THREAD

**Issue by leolll** [microsoft/clarity#345]
Things I've done to unmask form input fields:

1. Added a parent CSS class selector for the the form. (doesn't work)
2. Added `input` & `select` selectors as umasked (doesn't' work)
3. Added `data-clarity-unmask="true"` to a parent div of the form (doesn't work)

Is this expected behavior?

The only think I haven't tried is adding `data-clarity-unmask="true"` to each individual form input.  I'd have to modify the form plugin's code so I was trying to avoid that.

**Comment by ClaritySupport** [comment]
@leolll ,Content in the input boxes and drop-down menus will be masked in all [modes](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fclarity%2Fclarity-masking%23masking-modes&data=05%7C01%7Cv-vikasd%40microsoft.com%7C15d29986c341449a6a7c08db0e8bc729%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119765806157633%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=TlPYoM5EED%2BuKi6JlKxej7%2Fk%2Bw%2BW5GP%2FTRNO8FBkKWI%3D&reserved=0) and can't be customized. Please refer [Masking content | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fclarity%2Fclarity-masking&data=05%7C01%7Cv-vikasd%40microsoft.com%7C15d29986c341449a6a7c08db0e8bc729%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119765806157633%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=TY%2FnJV%2Blp6NQgsCSnXuy7Z2FvgzUjDPtpyws%2By%2FhMMA%3D&reserved=0) for more details.

**Comment by academialab** [comment]
> @leolll ,Content in the input boxes and drop-down menus will be masked in all [modes](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fclarity%2Fclarity-masking%23masking-modes&data=05%7C01%7Cv-vikasd%40microsoft.com%7C15d29986c341449a6a7c08db0e8bc729%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119765806157633%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=TlPYoM5EED%2BuKi6JlKxej7%2Fk%2Bw%2BW5GP%2FTRNO8FBkKWI%3D&reserved=0) and can't be customized. Please refer [Masking content | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fclarity%2Fclarity-masking&data=05%7C01%7Cv-vikasd%40microsoft.com%7C15d29986c341449a6a7c08db0e8bc729%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638119765806157633%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=TY%2FnJV%2Blp6NQgsCSnXuy7Z2FvgzUjDPtpyws%2By%2FhMMA%3D&reserved=0) for more details.

I think that it would be a good feature to allow unmask some input content on demand. We needed to review some inputs, and we ended up implementing a workaround to achieve the same result as if we had unmasked those inputs.
I mean, every project could calculate the impact of such a actions.

**Comment by leolll** [comment]
I agree, there are times when viewing a recording and seeing the inputs is very enlighting.  Especially if the user had validation errors and abandoned the form.

**Comment by shyamal890** [comment]
I understand credit card information being masked but why mask email, name, etc. Infact why mask input boxes at all. Doesn't make sense.

We want to be able to see text inputs for debugging purposes or for understanding who are dropping off our form and for what reason.

It would be great if Clarity can have a relook at this policy.

**Comment by peteralewis** [comment]
Utterly pointless - why mask **all** input fields and not allow them to be unmasked?!?! Doesn't make any sense whatsoever. The page the support post refers to above states: "The input box includes the following but not limited to...". What this should read is "all input boxes" and not bother listing specifics. It would be far better if it actually worked as per the documentation, so only those listed on that "help" page were forced masked and not optional.

**Comment by thebhalesh** [comment]
> We needed to review some inputs, and we ended up implementing a workaround to achieve the same result as if we had unmasked those inputs. I mean, every project could calculate the impact of such a actions.
> 
@academialab I'm in the same situation trying to review some input fields in clarity. What workaround did you implement? Please share. I'm not sure if you're allowed to share such workarounds here so can you please share it on telegram: @ bhalesh or facebook: @ bhlesh or email: bhalesh@ duck .com?

**Comment by tdowns** [comment]
@academialab - what is the workaround you implemented to see text entered? Our application has many filters and search boxes, and we are unable to understand exactly how a user is searching and using our product.

**Comment by peteralewis** [comment]
@tdowns I don't think you'll get a response as they didn't respond to the previous request for insight.

My guess is they either changed the field type, name or similar to bypass the masking, or if Clarity is masking every single input field however it's named, not named, type, etc, then they may have implemented a JS field replacement solution, whereby it uses a hidden input field to store the user response and simulates an input field via a p tag in a div or similar, capturing focus and user typed input via the div and displaying it in the p tag. Or something similar using Vue. 

A lot of effort to fix something broken in Clarity.

**Comment by San-Jeevan** [comment]
this makes the whole thing useless for us

**Comment by arecipe** [comment]
We would need this feature to track the customer reported issues with their addresses by unmasking some of the input fields, if it's by design what is your work around to troubleshoot?

**Comment by ClaritySupport** [comment]
It is by design and there is no work around.


Thanks,
Clarity Support Team
Visit our docs!: https://docs.microsoft.com/en-us/clarity
***@***.******@***.***>
***@***.***

Confidentiality note: This e-mail, and any attachment to it, contains privileged and confidential information intended only for the use of the individual(s) or entity named in the e-mail.  If the reader of the e-mail is not the intended recipient, or the employee or agent responsible for delivering it to the intended recipient, you are hereby notified that reading it is strictly prohibited.  If you have received this e-mail in error, please immediately return it to the sender and delete it from your system.

**Comment by arecipe** [comment]
By design how can we track a customer reported issue? How are we supposed to find the related recordings? I assume clarity is used for this purpose!

**Comment by ClaritySupport** [comment]
We don't support unmasking input fields as those are likely to contain sensitive data.

**Comment by peteralewis** [comment]
I think what Microsoft are trying to say, is that they are responsible across all countries for any sensitive data that will be recorded and stored. I appreciate it should be down to the developers choice and that they would be sensible enough to ensure nothing sensitive was trackable, but legally I think the responsibility belongs with Microsoft, which they're not prepared to accept.  Fair enough for a free product!

When clarity first released it used to have functionality where you could add an attribute that ensured a field didn't mask, this vaguely worked for a bit, but then was removed completely without giving a clear reason.

One way I've seen around it, is to use JS to mimic an input field, displaying it in a paragraph tag in a div box, in the same way you can style fields using fake replicas with the form input field hidden. That way Clarity won't mask it. It's a lot of work to bypass the issue though, although you could create a generic function that did this for all fields.

**Comment by nairmanu** [comment]
By default, Clarity masks all sensitive text, such as email addresses and info entered by users as mentioned in the Masking settings page. For more details Please refer [Masking content | Microsoft Learn](https://learn.microsoft.com/en-us/clarity/setup-and-installation/clarity-masking).
